tests/python_functional/src/syslog_ng_config/syslog_ng_config.py

```diff
@@ -114,12 +114,8 @@ def create_inner_logpath(self, statements=None, flags=None):
         inner_logpath = self.__create_logpath_with_conversion(statements, flags)
         return inner_logpath
 
-    def create_example_msg_generator(self, **options):
-        generator_source = SourceDriver(None)
-        generator_source.driver_name = "example_msg_generator"
-        generator_source.DEFAULT_MESSAGE = "-- Generated message. --"
-        generator_source.options = options
-        return generator_source
+    def create_example_msg_generator_source(self, **options):
```

Comment:
I still don't like these methods, it's just boilerplate, will be hard to maintain. Calling the constructor directly from tests is much better in my opinion. We'd see actual types this way and we could import them in tests.

Reply:
Hm, interesting. If I understand you correctly this would be your suggestion?

```python
from src.syslog_ng_config.statements.sources.file_source import FileSource
from src.syslog_ng_config.statements.destinations.file_destination import FileDestination

def test_acceptance(config, syslog_ng, input_log, expected_log, counter):
    file_source = FileSource(file_name="input.log")
    file_destination = FileDestination(file_name="output.log")
    config.create_logpath(statements=[file_source, file_destination])
    config.update_global_options(keep_hostname="yes")

    file_source.write_log(input_log, counter)
    syslog_ng.start(config)
    assert file_destination.read_logs(counter) == [expected_log] * counter
```

It could be work in this way, _but_ I have a plan to add some property-based tests which are highly depends on this form:
```python
def create_some_driver_name_source():
    pass

def create_some_driver_name_destination():
    pass
```

The plan was the following:
- SyslogNgConfig() class could be generate automatically driver objects (separately sources and destinations) by their names. (What else would be the scope of the SyslogNgConfig() class besides to generate driver objects?). 


If we have a plan to remove `create_...()` like functions we should replace it with a separate list of supported drivers. I mean the following:
```
class SyslogNgConfig():
     def __init__(self):
          ....
          self.supported_source_drivers = [FileSource, ExampleMessageGenerator]
          self.supported_destination_drivers = [FileDestination]
```
with this change I think the property-based tests could generate driver objects automatically from SyslogNgConfig() class.

After this what do you think?

Reply:
There are ways to get source/destination instances without writing boilerplate or maintaining lists, for example, registering your source/destination types into a registry (so you can get instances similar to the Service Locator pattern. This is an anti-pattern because it also hides dependencies, but it's slightly better than a manual list).

But OK, it is not the scope of this PR.

Reply:
Thank you for your suggestion. I will check that patterns.